pyccel/codegen/printing/fcode.py

```diff
@@ -1074,6 +1074,21 @@ def _print_NumpyArray(self, expr):
                 rhs_code = self._print(expr.arg)
         return rhs_code
 
+    def _print_NumpySign(self, expr):
+        str_x = self._print(expr.x)
+        kind  = self.print_kind(expr.x)
+
+        if expr.dtype == NativeComplex():
```

Comment:
I think this should be an identity comparison not just an equality so it would be better as:
```
expr.dtype is NativeComplex()
```
```
expr.dtype is NativeComplex()
```
or
```
isinstance(expr.dtype, NativeComplex)
```

Reply:
https://github.com/pyccel/pyccel/pull/1169/commits/8336ee19c4181a9f391a62810e67b4d179406bfa